spec/controllers/assignments_controller_spec.rb

```diff
@@ -25,11 +25,29 @@
   end
 
   describe 'POST #create', :vcr do
+    before do
+      request.env['HTTP_REFERER'] = 'http://test.host/classrooms/new'
+    end
+
     it 'creates a new Assignment' do
       expect do
         post :create, organization_id: organization.id, assignment: attributes_for(:assignment)
       end.to change { Assignment.count }
     end
+
+    context 'invalid starter_code format' do
+      it 'fails to create a new Assignment' do
+        expect do
+          post :create,
+               organization_id: organization.id,
+               assignment:      attributes_for(:assignment),
+               repo_name:       'https://github.com/rails/rails'
+        end.not_to change { Assignment.count }
+
+        expect(response).not_to have_http_status(:internal_server_error)
+        expect(flash[:error]).to eql('Invalid repository name, use the format owner/name')
```

Comment:
Generally I try to have one expectation per `it` block. You could rephrase this as three different cases:

* It does not create an assignment
* It does not return an http error
* It provides a friendly error message